go.sum

```diff
@@ -485,10 +485,8 @@ k8s.io/kube-openapi v0.0.0-20190709113604-33be087ad058 h1:di3XCwddOR9cWBNpfgXask
 k8s.io/kube-openapi v0.0.0-20190709113604-33be087ad058/go.mod h1:nfDlWeOsu3pUf4yWGL+ERqohP4YsZcBJXWMK+gkzOA4=
 k8s.io/repo-infra v0.0.0-20181204233714-00fe14e3d1a3 h1:WD6cPA3q7qxZe6Fwir0XjjGwGMaWbHlHUcjCcOzuRG0=
 k8s.io/repo-infra v0.0.0-20181204233714-00fe14e3d1a3/go.mod h1:+G1xBfZDfVFsm1Tj/HNCvg4QqWx8rJ2Fxpqr1rqp/gQ=
-k8s.io/utils v0.0.0-20190801114015-581e00157fb1 h1:+ySTxfHnfzZb9ys375PXNlLhkJPLKgHajBU0N62BDvE=
```

Comment:
This may be a silly question, but what's the best way to revert files back to HEAD in git without losing my current commits? Or is it better to just start again new?

Reply:
it looks like you have a mix of change commits and merge commits. I'd probably:
1. reset to the commit you started from (`git reset 46e58df837ad83febd3fa1330c07bc0d353d406f`)
2. stage/commit the changes you want to keep (discarding the go.mod,sum and vendor changes)
3. rebase on current master
4. re-pin the k8s.io/utils dependency

the result of steps 1 and 2 are at https://github.com/liggitt/kubernetes/commits/vegemitecheesetoast-master in case that helps

Reply:
When I commit and then i try to push, it forces me to do a pull to I guess pull in changes  that have happened since. Someone linked me documentation on how pin the new dependency, but I can't find it again. I tried to do step 4 based on the command I have run previously, but when I ran it, I got this error:

/coding/kubernetes$ ./hack/pin-dependency.sh k8s.io/utils v0.0.0-20190808222454-9c44e169899e
Running: go get k8s.io/utils@v0.0.0-20190808222454-9c44e169899e
go: malformed go.sum:
/home/xx/coding/kubernetes/go.sum:488: wrong number of fields 5
!!! Error in ./hack/pin-dependency.sh:65
  Error in ./hack/pin-dependency.sh:65. 'go get -m -d "${dep}@${sha}"' exited with status 1
Call stack:
  1: ./hack/pin-dependency.sh:65 main(...)
Exiting with status 1

That line is here:

k8s.io/utils v0.0.0-20190801114015-581e00157fb1/go.mod h1:sZAwmy6armz5eXlNoLmJcl4F1QuKu7sr+mFQ0byX7Ew=modernc.org/golex v1.0.0/go.mod h1:b/QX9oBD/LhixY6NDh+IdGv17hgB+51fET1i2kPSmvk=

I'm stuck on how to fix this!